1436790/1438462: Part 20: 20 files with MSDN link fixes.

DennisLee-DennisLee opened the pull request:
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

DennisLee-DennisLee committed:
20 more files with MSDN link fixes.

DennisLee-DennisLee committed:
Merge branch 'master' of https://github.com/dotnet/docs into v-dele-1438462-020

mairaw reviewed (changes requested):
Two suggestions and then we can merge. Thanks @DennisLee-DennisLee!

mairaw committed:
Apply suggestions from code review

Committing two of Maira's suggestions for previous-versions links that aren't attached to a docset.

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>
